client/src/Movies/Movie.js

```diff
@@ -26,16 +31,16 @@ export default class Movie extends Component {
       });
   };
   // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
+  componentWillReceiveProps(newProps){
```

Comment:
I just found out that `componentWillReceiveProps()` has been deprecated and there is a new pattern for this logic.
more info here >> https://reactjs.org/docs/react-component.html#updating
and here >> https://alligator.io/react/get-derived-state/